Dockerfile

```diff
@@ -15,52 +15,75 @@
 #
 
 FROM ubuntu:22.04 as tools-env
-ARG IMAGE_KERNEL_VERSION=v0.19.1
-ARG LINUX_VERSION=6.5.9-ctsi-1
+ARG IMAGE_KERNEL_VERSION=v0.20.0-output-unification-test1
+ARG LINUX_VERSION=6.5.13-ctsi-1-output-unification-test1
 ARG LINUX_HEADERS_URLPATH=https://github.com/cartesi/image-kernel/releases/download/${IMAGE_KERNEL_VERSION}/linux-libc-dev-riscv64-cross-${LINUX_VERSION}-${IMAGE_KERNEL_VERSION}.deb
 ARG BUILD_BASE=/opt/cartesi
 
 # Install dependencies
 # ------------------------------------------------------------------------------
 ENV LINUX_HEADERS_FILEPATH=/tmp/linux-libc-dev-riscv64-cross-${LINUX_VERSION}-${IMAGE_KERNEL_VERSION}.deb
 
-RUN DEBIAN_FRONTEND=noninteractive apt-get update && \
-    apt-get upgrade -y && \
-    apt-get install -y --no-install-recommends \
-        build-essential \
+RUN <<EOF
+export DEBIAN_FRONTEND=noninteractive
+apt-get update
+apt-get upgrade -y
+apt-get install -y --no-install-recommends \
+        dpkg-dev \
+        g++-12 \
+        gcc-12 \
+        make \
         ca-certificates \
         git \
         wget \
-        crossbuild-essential-riscv64 \
+        libclang-dev \
+        pkg-config \
+        dpkg-cross \
         gcc-12-riscv64-linux-gnu \
-        g++-12-riscv64-linux-gnu \
-        && \
-    wget -O ${LINUX_HEADERS_FILEPATH} ${LINUX_HEADERS_URLPATH} && \
-    echo "efdb2243d9b6828e90c826be0f178110f0cc590cb00e8fa588cb20723126c2a4  ${LINUX_HEADERS_FILEPATH}" | sha256sum --check && \
-    apt-get install -y --no-install-recommends ${LINUX_HEADERS_FILEPATH} && \
-    adduser developer -u 499 --gecos ",,," --disabled-password && \
-    mkdir -p ${BUILD_BASE}/tools && chown -R developer:developer ${BUILD_BASE}/tools && \
-    rm -rf /var/lib/apt/lists/* ${LINUX_HEADERS_FILEPATH}
+        g++-12-riscv64-linux-gnu
+
+for tool in cpp g++ gcc gcc-ar gcc-nm gcc-ranlib gcov gcov-dump gcov-tool; do
```

Comment:
Now instead of two loops we have `12` in multiple places, I like the other way better.
It also allowed switching between multiple versions such as `11` and `12` afterwards with little additional code.